Readme.md

```diff
@@ -1,12 +1,528 @@
+- applications status tracking
+
+1. [Install Garmadon-forwarder](#install-garmadon-forwarder) on your ResourceManager servers
+2. Deploy Garmadon-agent on your ResourceManager servers
+3. Update YARN_RESOURCEMANAGER_OPTS env variable in your ResourceManager start scripts with this property 
+ <pre>
+ -javaagent:<b><i>path-to-garmadon-agent.jar</i></b>=com.criteo.hadoop.garmadon.agent.modules.ResourceManagerModule
+ </pre>
+4. Restart all ResourceManagers
+
+#### [OPTIONAL] Activate Garmadon agent on any JVM
+
+Activating Garmadon-agent on a standalone JVM will provide:
+- HDFS interaction metrics realized outside of containers
+
+Since most of our data pipelines execute actions on HDFS directories outside of yarn containers, we force the use of Garmadon-agent on cluster gateway's by setting:
+<pre>
+JAVA_TOOL_OPTIONS=-javaagent:<b><i>path-to-garmadon-agent-jar</i></b>=com.criteo.hadoop.garmadon.agent.modules.StandaloneModule
+</pre>
+in users env.
+
+You can use any technique that fit your needs to load Garmadon-agent on any JVM that's performing HDFS calls with Hadoop library classes.
+
+**Just make sure to use com.criteo.hadoop.garmadon.agent.modules.StandaloneModule and to have a running forwarder beside the instrumented JVM.** 
+
+
+### Checking your install
+
+At this point, garmadon events should already feed your elasticsearch cluster (NodeManager's agent produces system metrics).
+
+Check that your NodeManager processes command line contains:
+```
+-javaagent:/usr/lib/hadoop-criteo/garmadon/lib/garmadon-agent.jar=com.criteo.hadoop.garmadon.agent.modules.NodeManagerModule
+```
+
+Check that your ResourceManager processes command line contains:
+```
+-javaagent:/usr/lib/hadoop-criteo/garmadon/lib/garmadon-agent.jar=com.criteo.hadoop.garmadon.agent.modules.ResourceManagerModule
+```
+
+Now, launch Yarn applications and see container metrics populate your dashboards.
+
+Your container's log should contain:
+
+```
+[main] INFO com.criteo.hadoop.garmadon.agent.EventAgent - Starting Garmadon Agent Version ....
+```
+
+## Main features
+
+### Dashboarding
+
+**All dashboards are based on realtime data generated by Garmadon.**
+
+_The following screenshots have high definition, don't hesitate to zoom-in_
+
+#### End users dashboards
+
+End users dashboard provide "per application" information
+
+##### Garmadon-yarn-application 
+
+This dashboard is applicable to all applications running on your cluster. For JVM based containers, you will benefit specific tab with JVM metrics.
+
+If you don't select a specific containers, the following panels will display Max value et average.
+That can be usefull if you want to check a threshold on a lot of containers. 
+
+For instance, what is the max heap occupancy across all mappers of a job ?
+
+**Make you sure your grafana time range includes data of the application you want to display !**
+**If not, the application will not be selectable in the drop-down list.**
```

Comment:
If the app id is still not here it is perhaps due to the max terms beeing taken from ES. In that you cas eyou can still type your app id manually